common/model/appUser.test.js

```diff
@@ -0,0 +1,164 @@
+          expect(res[0].active).toEqual(account1.data().active)
+          expect(res[0].email).toEqual(account1.data().email)
+          expect(res[0].hostname).toEqual(account1.data().hostname)
+          expect(res[0].key).toEqual(account1.data().key)
+          expect(res[0].name).toEqual(account1.data().name)
+          expect(res[0].updatedOn).toEqual(account1.data().updatedOn)
+
+          expect(res[1].username).toEqual(account2.data().username)
+          expect(res[1].active).toEqual(account2.data().active)
+          expect(res[1].email).toEqual(account2.data().email)
+          expect(res[1].hostname).toEqual(account2.data().hostname)
+          expect(res[1].key).toEqual(account2.data().key)
+          expect(res[1].name).toEqual(account2.data().name)
+          expect(res[1].updatedOn).toEqual(account2.data().updatedOn)
+
+          expect(res[2].username).toEqual(account3.data().username)
+          expect(res[2].active).toEqual(account3.data().active)
+          expect(res[2].email).toEqual(account3.data().email)
+          expect(res[2].hostname).toEqual(account3.data().hostname)
+          expect(res[2].key).toEqual(account3.data().key)
+          expect(res[2].name).toEqual(account3.data().name)
+          expect(res[2].updatedOn).toEqual(account3.data().updatedOn)
+
+          done()
+        })
+        .catch(done)
+    })
+  })
+
+  describe('the deleteAccount method', () => {
+    it('should delete the account of the user in database', (done) => {
+      user.deleteAccount(account1.id)
+        .then(res => {
+          expect(DB.connection).toHaveBeenCalledTimes(1)
+          expect(firestoreMock.mockCollection).toBeCalledWith('accounts')
+          expect(firestoreMock.mockCollection).toBeCalledWith('appUsers')
+          done()
+        })
+        .catch(done)
+    })
+  })
+
+  describe('the accountFromSnapshot method', () => {
+    it('should return a the data from the snapshot', (done) => {
+      const res = AppUser.accountFromSnapshot(account1)
+      expect(DB.connection).toHaveBeenCalledTimes(0)
+      expect(res.id).toEqual(account1.id)
+      expect(res.username).toEqual(account1.data().username)
+      expect(res.active).toEqual(account1.data().active)
+      expect(res.email).toEqual(account1.data().email)
+      expect(res.hostname).toEqual(account1.data().hostname)
+      expect(res.key).toEqual(account1.data().key)
+      expect(res.name).toEqual(account1.data().name)
+      expect(res.updatedOn).toEqual(account1.data().updatedOn)
+      done()
+    })
+  })
+
+  describe('the current method', () => {
+    it('should return the current user object', (done) => {
+      const user = AppUser.current()
+      expect(user.uid).toEqual('uid')
+      done()
+    })
+  })
+
+  describe('the subscribeAccounts method', () => {
+    it('should return the current user object', (done) => {
+      firestoreMock.mockOnSnaptshotSuccess = {
+        docs: [account1]
+      }
+      const test = (docs) => {}
```

Comment:
we should test that the result is the expected one,
in order to do so we can pass a `successCB` to `subscribeAccounts` and perform out checks in there, for example:
```
user.subscribeAccounts((res) => {
        expect(DB.connection).toHaveBeenCalledTimes(1)
        expect(firestoreMock.mockCollection).toBeCalledWith('accounts')
        expect(firestoreMock.mockCollection).toBeCalledWith('appUsers')
        expect(firestoreMock.mockOnSnaptshot).toHaveBeenCalledTimes(1)
        expect(res.length).toEqual(1)
        done()
      }, console.error)
```